### ISSUE TITLE
Option to suppress ignored file warnings

### ISSUE THREAD

**Issue by robertlevy** [eslint/eslint#6302]
Using version 2.10.2 with browserify via eslintify.  All files that our code 'require( )'s gets passed into eslint.  This includes things that should be ignored such as npm or bower packages.  Currently, this results in a bunch of warnings about ignored files which aren't really useful in this case.  Ideally, eslint would have an option to suppress ignored file warnings.   

Since eslintify is calling `executeOnText`, a `warnIgnored` parameter added here would probably be a quick fix: https://github.com/eslint/eslint/blob/c10c07f0eec529ea055e8d95d9f9899228274911/lib/cli-engine.js#L745

See also https://github.com/yaycmyk/eslintify/pull/10

**Comment by robertlevy** [comment]
See PR #6305

**Comment by vitorbal** [comment]
Hi @robertlevy! Thanks for the suggestion. Sounds solid, let's wait to see what the rest of the gang thinks :-)
Concerning your fix suggestion, I imagine we could get even more value out of this if the option can also be passed in to the CLI, something like this:

```bash
eslint --no-ignored-warnings --lib/**
```

What do you think?

**Comment by ilyavolodin** [comment]
Sounds reasonable to me. If we already support something in `executeOnFiles`, we should also support it in `executeOnText`. Only concern is, `executeOnFiles` calls into `executeOnText`. We need to make sure we don't get double the warnings in the case where things are ignored.

**Comment by platinumazure** [comment]
Any reason we can't move the warning for `executeOnFiles` into `executeOnText`? That would seem to be the simplest solution, from my limited perspective.

**Comment by robertlevy** [comment]
this PR simply skips the creation of warnings that were previously being generated, it shouldn't result in any new/duplicate warnings

**Comment by robertlevy** [comment]
any objection to merging this to unblock the eslintify issue and then creating another issue to track possibly adding a CLI flag and/or modifying executeOnFiles?

**Comment by ilyavolodin** [comment]
@robertlevy We need to hear from the rest of the team first, before marking this as accepted and merging PR. @eslint/eslint-team thoughts?

**Comment by nzakas** [comment]
Seems reasonable.
